library/Icingadb/Widget/Detail/UserDetail.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Widget\Detail;
+
+use Icinga\Module\Icingadb\Common\Auth;
+use Icinga\Module\Icingadb\Common\Links;
+use Icinga\Module\Icingadb\Model\User;
+use Icinga\Module\Icingadb\Widget\EmptyState;
+use Icinga\Module\Icingadb\Widget\HorizontalKeyValue;
+use Icinga\Module\Icingadb\Widget\ItemList\UsergroupList;
+use Icinga\Module\Icingadb\Widget\ShowMore;
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\Text;
+
+class UserDetail extends BaseHtmlElement
+{
+    use Auth;
+
+    /** @var User The given user */
+    protected $user;
+
+    protected $defaultAttributes = ['class' => 'user-detail'];
+
+    protected $tag = 'div';
+
+    public function __construct($user)
```

Comment:
The docblock for the `$user` property hints at `User`, so please also require the `$user` param to be an instance of that.